crown-court-contribution/src/main/java/uk/gov/justice/laa/crime/contribution/staticdata/enums/AppealContributionAmount.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.justice.laa.crime.contribution.staticdata.enums;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import uk.gov.justice.laa.crime.enums.AppealType;
+import uk.gov.justice.laa.crime.enums.AssessmentResult;
+import uk.gov.justice.laa.crime.enums.CrownCourtAppealOutcome;
+
+import java.math.BigDecimal;
+
+@Getter
+@AllArgsConstructor
+public enum AppealContributionAmount {
+
+    NO_CONTRIBUTION(new BigDecimal(0)),
+    PART_CONTRIBUTION(new BigDecimal(250)),
+    FULL_CONTRIBUTION(new BigDecimal(500));
+
+    private final BigDecimal contributionAmount;
+
+    public static AppealContributionAmount calculate(AppealType appealType, CrownCourtAppealOutcome appealOutcome, AssessmentResult assessmentResult) {
+        if (assessmentResult.equals(AssessmentResult.PASS)) {
+            return NO_CONTRIBUTION;
+        }
+
+        if (appealOutcome.equals(CrownCourtAppealOutcome.SUCCESSFUL)) {
+            return NO_CONTRIBUTION;
+        }
+
+        if ((appealType.equals(AppealType.ACS) && appealOutcome.equals(CrownCourtAppealOutcome.PART_SUCCESS))
+                || (appealType.equals(AppealType.ASE) && appealOutcome.equals(CrownCourtAppealOutcome.UNSUCCESSFUL))) {
+            return PART_CONTRIBUTION;
+        }
+
+        if (appealOutcome.equals(CrownCourtAppealOutcome.UNSUCCESSFUL)) {
+            return FULL_CONTRIBUTION;
+        }
+
+        return NO_CONTRIBUTION;
+    }
+}
```

Comment:
Was expecting the enum to have the values for outcome, assessment result etc and to then filter the correct value based on the arguments passed. However I think what you've done here is a lot clearer, nice 👍

Reply:
I was considering that (see the first commit description) but didn't really want to have 18 combinations defined (though it could be argued that is a simpler approach). I then thought about only defining the cases where contributions are non-zero (of which there are only four variants), but then I would have needed to define a zero contribution enum and at that point I would have struggled for sensible values for outcome, assessment result etc. 

There are certainly pros and cons either way, thanks for your feedback on this 🥇